### PR TITLE
Unbind overlays

### PR DESCRIPTION
**Pull request by javierarce**
The overlays weren't being unbinded, which could lead to several issues (like removing the overlay when pressing the backspace key).

Also: this PR only binds the `keyDown` event when is really required.

@xavijam: can you review this PR, please?

### CODE REVIEW

File: lib/assets/javascripts/cartodb/table/overlays/annotation.js [hunk 1]
```diff
@@ -159,6 +156,9 @@ cdb.admin.overlays.Annotation = cdb.geo.ui.Annotation.extend({
 
     this.killEvent(e);
 
+    $(document).unbind('keydown', this._onKeyDown);
+    $(document).bind('keydown', this._onKeyDown);
+
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/overlays/annotation.js]: Why do you need to do this?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2537/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2538/))

**Comment by xavijam** [comment]
@javierarce, let me know when changes are done please :)

**Comment by javierarce** [comment]
Done, @xavijam. I've removed the unnecessary "unbind" that you mentioned + fixed the failing spec.

**Comment by xavijam** [comment]
Ok, checking!

**Comment by xavijam** [comment]
As far as my knowledge goes, :+1:

**Pull request merged**